src/Common/tests/System/Net/Http/Http2LoopbackServer.cs

```diff
@@ -109,16 +109,21 @@ private async Task<bool> FillBufferAsync(Memory<byte> buffer, CancellationToken
         }
 
         public async Task<Frame> ReadFrameAsync(TimeSpan timeout)
+        {
+            using CancellationTokenSource timeoutCts = new CancellationTokenSource(timeout);
```

Comment:
won't this get immediately disposed and then you'd use token after dispose?

Reply:
New C# 8 feature - [using declaration](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-8#using-declarations).

Reply:
we may consider using Mono and older compilers...
(and bootstrapping new platforms with old tools)

Reply:
@stephentoub What is your opinion of this new C# feature?  There have a few PR's lately that introduced it into CoreFx codebase. But I'm not convinced it is making the codebase better.  It doesn't seem easier to read. And people ask questions all the time about this new syntax.

Reply:
I'm generally ok with it.  My only concern is if it ends up leading to not disposing of things as early as they otherwise would be, but that can be addressed on a case-by-case basis when it's an issue.  From a readability perspective, I personally like that it avoids lots of nesting that can be unnecessary.

Reply:
@wfurt Given the broad update for NRT and index/range, that hadn't seemed likely -- are there any plans for this?

Reply:
@davidsh needing to learn about new syntax isn't a good reason to keep a feature out.

In terms of readability, I don't really see any negatives here; it improves things by reducing the amount of indents. See e.g. `DbConnection` for a pathological case. Needing to pay attention to the containing scope rather than an explicitly defined scope has never been an issue in C++, though I'm sure we can think of abuse cases.

Reply:
We do not need to be concerned about compiling with an older compiler.  We already rely significantly in many places on having a C# 8 compiler.